src/domain/task-system/src/aggregates/task.rs

```diff
@@ -0,0 +1,276 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, Utc};
+use enum_variants::*;
+use event_sourcing::*;
+
+use crate::entities::*;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug)]
+pub struct Task {
+    state: TaskState,
+    pending_events: Vec<TaskEvent>,
+    last_synced_event: Option<EventID>,
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+impl Task {
+    pub fn new(task_id: TaskID, logical_plan: LogicalPlan) -> Self {
+        let genesis = TaskCreated {
+            event_time: Utc::now(),
+            task_id,
+            logical_plan,
+        };
+
+        Self {
+            state: TaskState {
+                task_id,
+                status: TaskStatus::Queued,
+                cancellation_requested: false,
+                logical_plan: genesis.logical_plan.clone(),
+                created_at: genesis.event_time.clone(),
+                ran_at: None,
+                cancellation_requested_at: None,
+                finished_at: None,
+            },
+            pending_events: vec![genesis.into()],
+            last_synced_event: None,
```

Comment:
Isn't first "TaskCreated"already synchronized to the state ?

Reply:
`last_synced_event` tracks last event that was persisted in the EventStore. Here we are creating a completely new task, so the `TaskCreated` should be considered a pending update.